src/test/benchmarks/Vector/shrink_and_clear.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// This benchmark checks the performance of the abstraction for its shrink and
+// truncate operations.
+include!{"../../rmc-prelude.rs"}
+
+fn operate_on_vec(times: usize) {
+    // Create vector with known capacity
+    let mut v: Vec<u32> = Vec::with_capacity(times);
+    for i in 0..times {
+        v.push(__nondet());
+    }
+    assert!(v.len() == times);
+    // Here, Vecs with grow() internally
+    v.push(1);
+    assert!(v.pop() == Some(1));
+    // Completely shrink the array to remove additional allocations
+    v.shrink_to_fit();
+    // Push some new elements to grow() again
```

Comment:
now check that the last element is right. Even better: take a nondet i, save its value, then check it hasn't changed